File: backend/src/CompanyProjectManagement.Domain/Entities/Empresa.cs
namespace CompanyProjectManagement.Domain.Entities;

public class Empresa
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Identificacion { get; set; } = string.Empty;
    public string Telefono { get; set; } = string.Empty;
    public string Direccion { get; set; } = string.Empty;
    public bool EstadoHabilitacion { get; set; } = true;
    public ICollection<Proyecto> Proyectos { get; set; } = new List<Proyecto>();
}

File: frontend/src/types/empresa.ts
export interface Empresa {
  id: number;
  nombre: string;
  identificacion: string;
  direccion: string;
  telefono: string;
  estadoHabilitacion: boolean;
  createdAt: string;
  updatedAt: string | null;
}

export interface CrearEmpresaRequest {
  nombre: string;
  identificacion: string;
  telefono: string;
  direccion: string;
  estadoHabilitacion?: boolean;
}

export interface ActualizarEmpresaRequest {
  nombre: string;
  identificacion: string;
  telefono: string;
  direccion: string;
  estadoHabilitacion: boolean;
}
